src/utils/prometheus.ts

```diff
@@ -0,0 +1,185 @@
+type ResponseSentParams = {
+  endpoint: string;
+  statusCode: string;
+  method: string;
+  duration: number;
+};
+
+type TransactionQueuedParams = {
+  chainId: string;
+  walletAddress: string;
+};
+
+type TransactionSentParams = TransactionQueuedParams & {
+  success: boolean;
+  duration: number;
+};
+
+type TransactionMinedParams = TransactionQueuedParams & {
+  endToEndDuration: number;
+  duration: number;
+};
+
+// Union type for all possible event parameters
+type MetricParams =
+  | { event: "response_sent"; params: ResponseSentParams }
+  | { event: "transaction_queued"; params: TransactionQueuedParams }
+  | { event: "transaction_sent"; params: TransactionSentParams }
+  | { event: "transaction_mined"; params: TransactionMinedParams };
+
+// Request metrics
+const requestsTotal = new Counter({
+  name: "engine_requests_total",
+  help: "Total number of completed requests",
+  labelNames: ["endpoint", "status_code", "method"],
+  registers: [enginePromRegister],
+});
+
+const requestDuration = new Histogram({
+  name: "engine_response_time_ms",
+  help: "Response time in milliseconds",
+  labelNames: ["endpoint", "status_code", "method"],
+  buckets: [100, 300, 500, 700, 1000, 3000, 5000, 7000, 10_000],
+  registers: [enginePromRegister],
+});
+
+// Transaction metrics
+const transactionsQueued = new Gauge({
+  name: "engine_transactions_queued",
+  help: "Number of transactions currently in queue",
+  labelNames: ["chain_id", "wallet_address"],
+  registers: [enginePromRegister],
+});
+
+const transactionsQueuedTotal = new Counter({
+  name: "engine_transactions_queued_total",
+  help: "Total number of transactions queued",
+  labelNames: ["chain_id", "wallet_address"],
+  registers: [enginePromRegister],
+});
+
+const transactionsSentTotal = new Counter({
+  name: "engine_transactions_sent_total",
+  help: "Total number of transactions sent",
+  labelNames: ["chain_id", "wallet_address", "is_success"],
+  registers: [enginePromRegister],
+});
+
+const transactionsMinedTotal = new Counter({
+  name: "engine_transactions_mined_total",
+  help: "Total number of transactions mined",
+  labelNames: ["chain_id", "wallet_address"],
+  registers: [enginePromRegister],
+});
+
+// Transaction duration histograms
+const queueToSentDuration = new Histogram({
+  name: "engine_transaction_queue_to_sent_seconds",
+  help: "Duration from queue to sent in seconds",
+  labelNames: ["chain_id", "wallet_address"],
+  buckets: [1, 5, 15, 30, 60, 120, 300, 600],
+  registers: [enginePromRegister],
+});
+
+const sentToMinedDuration = new Histogram({
+  name: "engine_transaction_sent_to_mined_seconds",
+  help: "Duration from sent to mined in seconds",
+  labelNames: ["chain_id", "wallet_address"],
+  buckets: [10, 30, 60, 120, 300, 600, 1200, 1800],
+  registers: [enginePromRegister],
+});
+
+const endToEndDuration = new Histogram({
+  name: "engine_transaction_end_to_end_seconds",
+  help: "Duration from request to mined in seconds",
```

Comment:
From queued to mined, since we aren't tracking "request to queued" (the logic in `mineTransactionWorker.ts` records `minedAt` - `queuedAt`)